### PR TITLE
[#176458169] Remove unused code path for bonus activation

### PR DESCRIPTION
**Pull request by balanza**
#### Description
As Bonus Vacanze cannot be requested anymore, we can get rid of all the code related su such features:
* `BONUS_REQUEST_LIMIT_DATE` env var in configuration
* Endpoints to always return `404`
* unused code to be removed

#### Motivation and Context
There is a known issue on the infrastructure: when applying a change, environment variables which are date string, are converted unexpectedly. UTC ISO date format is exposed to such issue, thus `BONUS_REQUEST_LIMIT_DATE` may be in an unexpected format causing the application boot to fail.

#### How Has This Been Tested?
Locally using `io-mock`

#### Types of changes

- [x] Chore
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by balanza** [comment]
This is actually an alternative for https://github.com/pagopa/io-backend/pull/754

File: api_bonus.yaml [hunk 1]
```diff
@@ -71,39 +45,9 @@ paths:
...
-            Location:
-              type: string
-              description: |-
-                Location (URL) of created request resource.
-                A GET request to this URL returns the request status and details.
-        "202":
-          description: Processing request.
-          schema:
-            $ref: "#/definitions/InstanceId"
-        "401":
```
Comment (BurnedMarshal) [inline api_bonus.yaml:88]: this API could return 401 with `session_token` evaluation (`bearerSessionTokenAuth`) or 500 if something goes wrong with redis client

File: src/utils/responses.ts [hunk 2]
```diff
@@ -26,6 +27,14 @@ export function ResponseNoContent(): IResponseNoContent {
   };
 }
 
+/**
+ * A response error for when a feature has been dismissed.
+ */
+export const ResponseErrorDismissed = ResponseErrorNotFound(
```
Comment (BurnedMarshal) [inline src/utils/responses.ts:33]: this response type should be shared with the App. Maybe must be included in `io-ts-commons`?

File: src/config.ts [hunk 3]
```diff
@@ -324,17 +323,6 @@ export const BONUS_API_CLIENT = BonusAPIClient(
   BONUS_API_URL,
   httpApiFetch
 );
-// the date until a user can request a bonus
-export const BONUS_REQUEST_LIMIT_DATE = UTCISODateFromString.decode(
-  process.env.BONUS_REQUEST_LIMIT_DATE
```
Comment (BurnedMarshal) [inline src/config.ts:329]: Reference to `BONUS_REQUEST_LIMIT_DATE` should be removed from README end `.env.example`

File: api_bonus.yaml [hunk 4]
```diff
@@ -14,34 +14,8 @@ paths:
...
-            $ref: "#/definitions/InstanceId"
-          headers:
-            Location:
-              type: string
-              description: |-
-                Location (URL) of created request resource.
-                A GET request to this URL returns the request status and details.
-        "202":
-          description: Processing request.
-        "401":
```
Comment (BurnedMarshal) [inline api_bonus.yaml:29]: this API could return 401 with session_token evaluation (bearerSessionTokenAuth) or 500 if something goes wrong with redis client

**Review by BurnedMarshal — CHANGES_REQUESTED** [review]

File: api_bonus.yaml (hunk 4 above)
Comment (balanza) [reply api_bonus.yaml:29]: You're right! My bad.

Question: should we evaluate the token as well? Shouldn't we just return "dismissed"?

File: src/utils/responses.ts (hunk 2 above)
Comment (balanza) [reply src/utils/responses.ts:33]: We definitely want to map more specific response like this in `io-ts-commons`. However, there's no point for this scenario as we aren't mapping anything, we are just passing a descriptive string.

File: api_bonus.yaml (hunk 4 above)
Comment (BurnedMarshal) [reply api_bonus.yaml:29]: The API definition requires an Authorization header. If the token is invalid maybe is better returns the proper error evaluating it.

File: api_bonus.yaml [hunk 5]
```diff
@@ -14,34 +14,8 @@ paths:
...
-          schema:
-            $ref: "#/definitions/ProblemJson"
-        "451":
-          description: Unauthorized for legal reason.
-        "500":
-          description: Service unavailable.
-          schema:
-            $ref: "#/definitions/ProblemJson"
+        "404":
+          description: The feature has been dismissed
```
Comment (BurnedMarshal) [inline api_bonus.yaml:18]: Maybe we should add the response schema to keep compliant the spec with the real response provided by the API
```suggestion
          description: The feature has been dismissed
          schema:
            $ref: "#/definitions/ProblemJson"
```

**Review by Undermaken — APPROVED** [review]

**Review by BurnedMarshal — APPROVED** [review]
lgtm

**Pull request merged**